Source/WebCore/platform/graphics/ComplexTextController.cpp

```diff
@@ -378,6 +392,11 @@ void ComplexTextController::collectComplexTextRuns()
 
         nextFont = m_font.fontForCombiningCharacterSequence(baseOfString.subspan(previousIndex, currentIndex - previousIndex));
 
+        if (shouldProcessTextSpacingTrim && nextFont && !nextFont->isSystemFontFallbackPlaceholder()) {
```

Comment:
We use this conditional twice which makes me think we could create some of helper function to refactor it into. That would probably also help us give a meaningful name for this logic to help improve the readability imo